### PR TITLE
Get random region for Gardener cluster in backend switching job and check for is-PR-approved

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Assigns random region for Gardener cluster in backend switching job.

**Related issue(s)**

- https://github.com/kyma-project/eventing-manager/issues/131

### CODE REVIEW

File: .github/workflows/e2e-backend-switching-reuseable.yml [hunk 1]
```diff
@@ -60,6 +58,9 @@ jobs:
           # generate cluster name and export it to Github env for cleanup step to access it.
           export CLUSTER_NAME="${CLUSTER_PREFIX}$(openssl rand -hex 2)"
           echo "CLUSTER_NAME=${CLUSTER_NAME}" >> $GITHUB_ENV
+          # set random region for AWS.
+          export GARDENER_REGION=$(./scripts/gardener/aws/get_random_region.sh)
+          export GARDENER_ZONES="${GARDENER_REGION}a"
```
Comment (muralov) [inline .github/workflows/e2e-backend-switching-reuseable.yml:63]: why do we always use zones with the suffix `a``

File: .github/workflows/pull-target-e2e.yml [hunk 2]
```diff
@@ -17,8 +17,19 @@ on:
...
+      - name: Check if PR approved
+        env:
+          GH_TOKEN: ${{ github.token }}
+          PR_URL: https://github.com/kyma-project/eventing-manager/pull/${{ github.event.number }}
+        run: |
+          ./scripts/is_pr_approved.sh
+
   wait-until-build-succeeds:
     runs-on: ubuntu-latest
+    needs: is-pr-approved
```
Comment (muralov) [inline .github/workflows/pull-target-e2e.yml:32]: why don't you want to run this if an PR not approved?

File: .github/workflows/e2e-backend-switching-reuseable.yml (hunk 1 above)
Comment (mfaizanse) [reply .github/workflows/e2e-backend-switching-reuseable.yml:63]: We are always using the first zone in the region, we do not have any preference for zones.

File: .github/workflows/pull-target-e2e.yml (hunk 2 above)
Comment (mfaizanse) [reply .github/workflows/pull-target-e2e.yml:32]: We have quota limits on number of gardener clusters, so we are filtering out to only run on approved PRs so less number of gardener clusters are created at a time.

**Review by muralov — APPROVED** [review]

**Pull request merged**